extensions/resource-deployment/src/ui/toolsAndEulaSettingsPage.ts

```diff
@@ -0,0 +1,516 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { EOL } from 'os';
+import * as nls from 'vscode-nls';
+import { AgreementInfo, DeploymentProvider, ITool, ResourceType, ResourceTypeOptionValue, ToolStatus } from '../interfaces';
+import { createFlexContainer } from './modelViewUtils';
+import * as loc from '../localizedConstants';
+import { IToolsService } from '../services/toolsService';
+import { getErrorMessage } from '../common/utils';
+import { ResourceTypePage } from './resourceTypePage';
+import { ResourceTypeWizard } from './resourceTypeWizard';
+
+const localize = nls.loadMessageBundle();
+
+export class ToolsAndEulaPage extends ResourceTypePage {
+	private form!: azdata.FormBuilder;
+	private view!: azdata.ModelView;
+	private toolRefreshTimestamp: number = 0;
+	private _optionsContainer!: azdata.FlexContainer;
+	private _toolsTable!: azdata.TableComponent;
+	private _optionDropDownMap: Map<string, azdata.DropDownComponent> = new Map();
+	private _toolsLoadingComponent!: azdata.LoadingComponent;
+	private _agreementContainer!: azdata.DivContainer;
+	private _agreeementCheckBox!: azdata.CheckBoxComponent;
+	private _installToolButton!: azdata.ButtonComponent;
+	private _installationInProgress: boolean = false;
+	private _tools: ITool[] = [];
+	private _eulaValidationSucceeded: boolean = false;
+	private _isInitialized = false;
+	private _isDoneButtonEnabled = false;
+	private _resourceType: ResourceType;
+
+	public set resourceProvider(provider: DeploymentProvider) {
+		this.wizard.provider = provider;
+	}
+
+	public get toolsService(): IToolsService {
+		return this.wizard.toolsService;
+	}
+
+	constructor(wizard: ResourceTypeWizard) {
+		super(localize('notebookWizard.toolsAndEulaPageTitle', "Deployment pre-requisites"), '', wizard);
+		this._resourceType = wizard.resourceType;
+	}
+
+	public async onEnter(): Promise<void> {
+		this.wizard.wizardObject.generateScriptButton.hidden = true;
+		this.wizard.wizardObject.registerNavigationValidator(async (pcInfo) => {
+			if (pcInfo.newPage < pcInfo.lastPage) {
```

Comment:
Add a comment explaining what this check is doing.

Reply:
Unnecessary check. Removing it.